### PR TITLE
Use LLVM Bazel BUILD files from google/llvm-bazel

### PR DESCRIPTION
**Pull request by GMNGeoffrey**
Adopts build files from https://github.com/google/llvm-bazel.

This is another attempt at https://github.com/google/iree/pull/2794 now
that https://github.com/google/llvm-bazel/pull/60 added Windows
support.

These make our LLVM build independent of TensorFlow. Switching to them
does slightly increase maintenance burden on our team because the TF
team is not assisting in maintaining them. It is not a guarantee that
llvm-bazel will already have a commit containing the up to date
workflows. However, it does have an advertised tag for each LLVM commit
that is the current best-effort for what builds, a workflow to
automatically pull in new LLVM commits and a build pipeline that
notifies me as soon as the build breaks (independent of any integration
into Google source control). I've been mostly able to stay ahead of the
integrates into Google source control.

I will be expanding our build cop docs to cover more of the workflows
here as I am buildcopping this week, but the general workflow is to run
`./scripts/git/update_llvm_dependent_submodules.sh` in lieu of the
update old `update_tf_submodule.sh` script. A GitHub action will also
create those PRs automatically to be merged.

### CODE REVIEW

**Review by stellaraccident — APPROVED** [review]
Very nice!

**Comment by GMNGeoffrey** [comment]
Thanks, but this needs https://github.com/google/llvm-bazel/pull/60 if you want to look at that :grin:

**Comment by GMNGeoffrey** [comment]
@ScottTodd I've built this on windows. Can you also give it a try and make sure it works for you?

File: .gitmodules [hunk 1]
```diff
@@ -59,3 +59,6 @@
 [submodule "third_party/cpuinfo"]
 	path = third_party/cpuinfo
 	url = https://github.com/pytorch/cpuinfo.git
+[submodule "third_party/llvm-bazel"]
+	path = third_party/llvm-bazel
+	url = https://github.com/GMNGeoffrey/llvm-bazel.git
```
Comment (ScottTodd) [inline .gitmodules:64]: Do you still want this to point to your fork?

**Review by ScottTodd — COMMENTED** [review]
Trying this now.

File: .gitmodules (hunk 1 above)
Comment (GMNGeoffrey) [reply .gitmodules:64]: OOoh no. I thought I fixed that!

**Comment by ScottTodd** [comment]
> @ScottTodd I've built this on windows. Can you also give it a try and make sure it works for you?

Sure :D. On my machine, the builds I care about continue to work at this PR.

`iree/tools/...` builds:
```powershell
D:\dev\projects\iree (pr/GMNGeoffrey/3544 -> GMNGeoffrey)
λ bazel build iree/tools/...
...
INFO: Elapsed time: 221.744s, Critical Path: 110.06s
INFO: 2521 processes: 2521 local.
INFO: Build completed successfully, 2636 total actions
```

Colab builds:
```powershell
D:\dev\projects\iree (pr/GMNGeoffrey/3544 -> GMNGeoffrey)
λ python3 .\colab\start_colab_kernel.py
...
Target //colab:everything_for_colab up-to-date:
  bazel-bin/colab/everything_for_colab.exe
  bazel-bin/colab/everything_for_colab.zip
INFO: Elapsed time: 155.824s, Critical Path: 113.19s
INFO: 773 processes: 773 local.
INFO: Build completed successfully, 778 total actions
...
[I 09:17:52.075 NotebookApp] Serving notebooks from local directory: D:\dev\projects\iree

**Review by ScottTodd — APPROVED** [review]

**Pull request merged**